ai/core/src/unitycatalog/ai/core/executor/local_subprocess.py

```diff
@@ -0,0 +1,154 @@
+import ast
+import base64
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import textwrap
+from typing import Any
+
+import cloudpickle
+
+from unitycatalog.ai.core.envs.executor_env_vars import (
+    EXECUTOR_MAX_CPU_TIME_LIMIT,
+    EXECUTOR_MAX_MEMORY_LIMIT,
+    EXECUTOR_TIMEOUT,
+)
+from unitycatalog.ai.core.executor.common import (
+    MB_CONVERSION,
+    NO_OUTPUT_MESSAGE,
+    TIMEOUT_ERROR_MESSAGE,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _extract_function_name(function_source: str) -> str:
+    """
+    Extracts the name of the first function defined in the provided source code
+    using the ast module.
+    """
+    try:
+        module_ast = ast.parse(textwrap.dedent(function_source))
+    except SyntaxError as e:
+        raise ValueError("Invalid Python source code") from e
+
+    for node in module_ast.body:
+        if isinstance(node, ast.FunctionDef):
+            return node.name
+    raise ValueError("No function definition found in the provided source code.")
+
+
+def _generate_runner_script(function_source: str, cpu_time_limit: int, memory_limit: int) -> str:
```

Comment:
Instead of this long string, I think we can create a separate python file for this, and pass those parameters as args. But this can be a follow-up :)